scanspec/regions.py

```diff
@@ -39,7 +42,7 @@ def axis_sets(self) -> List[Set[str]]:
         this region spans"""
         raise NotImplementedError(self)
 
-    def mask(self, points: AxesPoints) -> np.ndarray:
+    def mask(self, points: AxesPoints[K]) -> np.ndarray:
```

Comment:
Do we need to make Region a generic too?

Reply:
I think so, I just pushed it, let me know if I missed anywhere.